### PR TITLE
[Validator] pass error code as a string

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
What's the goal of this change?

File: src/Symfony/Component/Validator/Tests/Validator/AbstractTest.php [hunk 1]
```diff
@@ -528,7 +528,7 @@ public function testAddCustomizedViolation()
         $this->assertSame($entity, $violations[0]->getRoot());
         $this->assertSame('Invalid value', $violations[0]->getInvalidValue());
         $this->assertSame(2, $violations[0]->getPlural());
-        $this->assertSame(42, $violations[0]->getCode());
+        $this->assertSame('42', $violations[0]->getCode());
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Validator/AbstractTest.php]: The argument is documented as `string` in `ConstraintViolationBuilderInterface::setCode()`.  When adding type-hints in 5.0, we will need to compare the returned code with the string `'42'`. Making the change in the test here makes it consistent across all branches.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

File: src/Symfony/Component/Validator/Tests/Validator/AbstractTest.php (hunk 1 above)
Comment (Tobion) [reply src/Symfony/Component/Validator/Tests/Validator/AbstractTest.php]: ref. https://github.com/symfony/symfony/pull/32066#discussion_r294106130

**Pull request merged**